### ISSUE TITLE
Add the ability to tag blockchain accounts

### ISSUE THREAD

**Issue by chevdor** [rotki/rotki#461]
## Abstract

Currently, you add an account solely by this address.

## Motivation

Unless you know all your address (which is likely not the case) you may have a hard time remembering where an account comes from. Is that a hardware wallet? An app on your phone? Some metamask somewhere, ...

## Specification

When Adding a blockchain account, I suggest adding an option field for the name of that account. Improving further, the ability to tag accounts would likely be very handy.

**Comment by LefterisJP** [comment]
Hey @chevdor! Long time no see!

That's a good feature request and I agree it is needed.

At the moment @kelsos is working on the front-end vue-rewrite to make maintaining the front-end easier. After that this can definitely be worked on.

**Comment by chevdor** [comment]
Hi @LefterisJP, definitely still around :)
If you make it to web3 summit in Berlin, we could meet there.

Great news to hear that the UI will get some love. I see in other issues that I am reporting whatever friction I notice as 'first time user'. I hope that can help @kelsos for the new version.

I think Tagging helps in general users to organize themselves in an elegant way. I would even argue that once tagging is available, naming an account becomes less of an issue if not present.

**Comment by kelsos** [comment]
It should be way easier to add features to the UI after we are done with the rewrite.

**Comment by LefterisJP** [comment]
When tags are introduced I am wondering if the application should start with a default pre-created set of tags? @chevdor 

If yes what should they be? I am thinking:

- mining
- hardware wallet
- cold wallet
- paper wallet
- mobile wallet

Too much? Too little? Something missing? Naturally we will allow users to add/remove tags but a good set of defaults is important I think.

**Comment by chevdor** [comment]
Tags are not exclusive so setting default ones if you can make a good bet never hurst IMO.
For instance, for a Coinbase wallet, you may tag Coinbase. That's a safe bet :)
For a Bitcoin wallet, you may also tag Bitcoin. I would not add the list you suggest as you dont know if people need them and if they call it like that.

As far as defining the list, I would not. The reasons are:
- no one uses the same
- some users may localize their tags

What is important is to be case insentive to avoid duplicates such as Coinbase and coinbase.
It is also important to create tags when users use a string never seen before.

**Comment by LefterisJP** [comment]
> What is important is to be case insentive to avoid duplicates such as Coinbase and coinbase.

Case insensitive ... huh this is gonna suck. I can see the benefit but damn ...

> It is also important to create tags when users use a string never seen before.

What do you mean by that?

> I would not add the list you suggest as you dont know if people need them and if they call it like that.

That's true. But I just want to have something to show the new feature to the users. Few people read docs, fewer the changelog. So I don't expect many people to know this exists. But if they see a dropdown and click it next to their accounts and see already some tags with the ability to create new ones (or remove the old ones) then perhaps they will catch the drift immediately.

**Comment by chevdor** [comment]
> It is also important to create tags when users use a string never seen before.
> What do you mean by that?

You can check out here what has been done: https://polkadot.js.org/apps/#/accounts
It works pretty good.

There is a tag field and users can type in. It does a search and shows existing tags (autocomplete) or if you press enter and the string is unknown it creates a new tag.

> I would not add the list you suggest as you dont know if people need them and if they call it like that.

Then I would only add Bitoin, Ethereum, Paper Wallet and Mobile Wallet as example. The thing is: if you define some defaults, you need to let the user 'manage' their tags and allow them to delete some. For instance you call is `Hardware Wallet`, I would likely tag it `HW`, some users will call it `Secure`or `Schatz`. Everyone is different....

**Comment by LefterisJP** [comment]
> You can check out here 

Thanks a lot, will have a look.

> Then I would only add Bitoin, Ethereum, Paper Wallet and Mobile Wallet as example. The thing is: if you define some defaults, you need to let the user 'manage' their tags and allow them to delete some. For instance you call is Hardware Wallet, I would likely tag it HW, some users will call it Secureor Schatz. Everyone is different....

Yep I agree. I imagine we will let the user manage tags. Essentially create/delete/rename tags.

@kelsos Please also have a look and chime in if there are any gotchas or things to take care of front-end side.

TL;DR for the UI as I imagine it:

- Each blockchain account (ETH/BTC whatever) will also have a label field, which will by default be `None` and not be shown at all. So just the address is shown.
- If label is not None, instead of the address the label should be shown and somewhere the address to which this label corresponds to should be visible. Perhaps tooltip?
- A user should be able to add/remove a label from an account
- Each blockchain account should also be able to accept multiple tags. What kind of UI component would show the multiple tags next to the account address and label best?
- Users can add/remove any amount of tags they want from a blockchain account
- A tag is comprised of an all lowercase string and a description.
- There should be another component (or well any place you also think would look nice) where users can manage their exisiting tags. Either create new ones, remove or rename existing ones.

**Comment by chevdor** [comment]
Instead of a description (or additionally), you could plan a background color and a foreground color.
You would not have to manage it right away in the UI but I find labels easier to organise when they come with colors. For instance, one may use green for all crypto names such as Bitcoin or BTC, Ethereum or ETH, and other colors for other meanings.

**Comment by LefterisJP** [comment]
Yeah colors are also a good idea. Thought of adding it but maybe not in the first iteration. Depends on what @kelsos says

**Comment by chevdor** [comment]
My take would be: plan in the backend asap (=now), include in the frontend later if that helps.

**Comment by LefterisJP** [comment]
Yeah it's already in the DB table as RGB color format (TEXT in sqlite)

**Comment by kelsos** [comment]
Sounds good with the colors too. Vuetify should have a few components already with this kind of functionality but I don't think it would be that much of a hustle to implement them from scratch if needed.
